registry/storage/garbagecollect.go

```diff
@@ -122,6 +139,14 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 			if err != nil {
 				return fmt.Errorf("failed to delete manifest %s: %v", obj.Digest, err)
 			}
+			for _, layerDgst := range obj.Layers {
+				if _, ok := markSet[layerDgst]; !ok {
+					err := vacuum.RemoveLayerLink(obj.Name, string(layerDgst))
```

Comment:
I see we're casting the `layerDgst` to a string here, then parsing it back into a `digest.Digest` in the vacuum. Could we pass it as a `digets.Digest` to avoid the round-trip?